sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -63,6 +63,14 @@ struct _pi_object {
   std::atomic<pi_uint32> RefCount;
 };
 
+// Record for a memory allocation. This structure is used to keep track of all
+// memory allocations.
+struct MemAllocRecord : _pi_object {
+  MemAllocRecord(pi_context Context, void *Ptr) : Ptr(Ptr), Context(Context) {}
+  void *Ptr;
+  pi_context Context;
```

Comment:
why do you need to save Context?

Reply:
OK. can you add a TODO that this should go away with memory isolation fixed.

Reply:
Added TODO.